Metadata/AmazonMetadataBuilder.php

```diff
@@ -13,7 +13,7 @@
 
 use Aws\S3\Enum\CannedAcl;
 use Aws\S3\Enum\Storage;
-use Guzzle\Http\Mimetypes;
+use Guzzle\Http\Mimetypes; # TODO: Find something common
```

Comment:
@core23 is this a package of its own? guzzle/mimetypes or something? if so i think its ok to depend on that specifically and use it here. no need to abstract everything. the mimetype handling has no outside dependencies or influences or configuration like the http clients do.